doc/rst/technotes/libraries.rst

```diff
@@ -111,23 +205,145 @@ example looks like this:
+
+We don't anticipate supporting ``python2`` at this time, sorry.
+
+Compiling Your Chapel Library
+-----------------------------
+
+In addition to compiling with ``--library``, it is recommended to compile with
+``--library-python``.  This will create a Python-compatible module in addition
+to the normally generated library and header file.
+
+Python Module Name
+------------------
+
+By default, the name of the generated Python module will match the base name
+of the generated library.  In the ``foo.chpl`` example, this would mean the
+Python module will be named ``foo``.
+
+The name of the generated module can be changed independently of the generated
+library name using the compilation flag ``--library-python-name``.
+
+PYTHONPATH
+----------
+
+To use your library in a Python program, you will need to extend your
+``PYTHONPATH`` environment variable to include the directory where your library
+files are generated, e.g.:
+
+.. code-block:: sh
+
+   export PYTHONPATH=lib/:$PYTHONPATH
+
+See :ref:`Location of the Generated Library` for where your library files are
+generated, and how to change this location when compiling your Chapel library.
+
+Initializing and Using Your Library in Python
+---------------------------------------------
+
+Once your ``PYTHONPATH`` is set up and the Python module created, you can
+``import`` the module like a normal Python module.
+
+Similarly to using your library with C, you will need to call a set up function
+to ensure the Chapel runtime and standard modules are initialized, as well as
+a clean up function.
+
+Unlike the C case, the set up function is called ``chpl_setup()`` and will also
+handle initializing your module.   This function will still need to be called
+prior to any Chapel library function calls.
+
+Also unlike the C case, the clean up function is called ``chpl_cleanup()``.
+This function will still need to be called after all the Chapel library function
+calls are finished.
+
+For example:
+
+.. code-block:: Python
+
+   import foo
+
+   foo.chpl_setup()
+
+   foo.baz(7) // Call into a library function
+
+   foo.chpl_cleanup()
+
+.. note::
+
+   Currently the ``chpl_cleanup()`` function will also cause the Python program
+   to exit.  This may change in the future, but for now please ensure your
+   Python functionality is also complete before calling this function.
+
+In Case Something Goes Wrong
+----------------------------
+
+This compilation strategy uses Cython under the covers, generating a
+``chpl_foo.pxd`` file, a ``foo.pyx`` file, and a ``foo.py`` file by default for
+a ``libfoo.a`` / ``libfoo.so``, which are then called using a Cython command
+(this command is rather long due to the need to include the Chapel runtime and
+third-party libraries).  These files are currently left in the same location as
+the generated library - if compilation fails due to generating one or more of
+these files incorrectly, you may be able to modify the file and re-run the
+Cython command yourself.
```

Comment:
Would be nice to have an example of this, if feasible.

Reply:
I thought about it, but the example is going to be huge and difficult to read.  Plus, any error they get from Cython will tell them the Cython command used.  Should I point that out?

Reply:
> but the example is going to be huge and difficult to read

Ah, OK. That's why I said "if feasible" :)

> Plus, any error they get from Cython will tell them the Cython command used. Should I point that out?

Not sure how useful that would be. I'll leave it up to your judgement.